porch/controllers/packagevariants/pkg/controllers/packagevariant/packagevariant_controller.go

```diff
@@ -768,3 +774,57 @@ func ensurePackageContext(pv *api.PackageVariant,
 	prr.Spec.Resources["package-context.yaml"] = cm.String()
 	return nil
 }
+
+func ensureKRMFunctions(pv *api.PackageVariant,
+	prr *porchapi.PackageRevisionResources) error {
+
+	if len(pv.Spec.Mutators) == 0 {
+		return nil
+	}
+
+	if _, ok := prr.Spec.Resources[kptfilev1.KptFileName]; !ok {
+		return fmt.Errorf("%s not found in PackageRevisionResources '%s/%s'", kptfilev1.KptFileName, prr.Namespace, prr.Name)
+	}
+
+	// parse PackageVariant Mutators
+	mpvmutators, err := kyaml.Marshal(pv.Spec.Mutators)
```

Comment:
I don't think we need to do this. I think we can just create a list of functions (typed go struct) for each of the mutators and validators as specified in the design, based on the Go structs here. The pv here was loaded from the API server - there are no comments or anything. If there are any comments to preserve they are the ones in the PRR Kptfile.

So, once we have a complete list of "our" mutators, as go structs, we then look at the pipeline.mutators field in the Kptfile KubeObject and get a slice of subobjects back (see https://pkg.go.dev/github.com/GoogleContainerTools/kpt-functions-sdk/go/fn#SubObject.NestedSlice).

If our list of structs is "ourMutators", then something like:

```
 var newMutators fn.SliceSubObjects
 for _, f := range ourMutators {
  ko, err := fn.NewFromTypedObject(&f)
  if err != nil {
    return err
   }
  newMutators = append(newMutators, ko)
}
existingMutators := kptfileKubeObject.NestedSlice([]string{"pipeline", "mutators"})
 for _, ko := range existingMutators {
  name, ok, err := ko.NestedString("name")
// check ok, err
 if name is one of our names continue
 newMutators = append(newMutators, ko)
}
// then SetSlice with the newMutators
```